### DISCUSSION TITLE
Use TCP listener to send initial socket connection request to an Analyzer

### DISCUSSION THREAD

**Discussion by hamzabawumia** [nextgenhealthcare/connect#6100] [General]
I have a problem, I am using mirth connect on my computer. I understand that BC 5150 has the port set to port 5100 and the Analyzer always act as the server while LIS PC should act as a client.

Now how do i send an initial socket connection request from the Mirth TCP listener to the Analyzer so that the analyzer now knows where to send its results.

I have some java code to establish the connection but where do i put it? and how will the analyzer know which port to client is listening on ?

```
import java.io.*;
import java.net.*;

public class MyClient {
    public static void main(String[] args) {
        try {
            String AnalyzerIP = "192.168.1.100";
            int AnalyzerPort = 5150;

            Socket socket = new Socket(AnalyzerIP, AnalyzerPort);

            // Now you have a socket connection to your Mirth server!
            // You can send data over this socket.

            // Close the socket when done:
            socket.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

```
from stackoverflow (https://stackoverflow.com/questions/63709033/where-to-set-mindray-bc-5150-hl7-communication-port-number-and-ip-address
) there is a comment that states that ; "Once the handshake is completed, the server will push the results through that socket to the client pc. " But how does the server know which port the client pc is listening on?

This is quite confusing to me. I have also worked with other analyzers before, and the setting was simple since the Analyzer had an interface to enter the exact LIS ip address and port number to send results.

**Comment by hamzabawumia** [comment]
So I mistakenly posted this issue as a bug and @jonbartels solved the issue by suggesting that I setup a TCP Listener running in "client mode"  and it worked.

Here is the link he pointed me to: https://docs.nextgen.com/bundle/Mirth_User_Guide_41/page/connect/connect/topics/c_TCP_Listener_connect_ug.html

**Reply by jonbartels** [reply]
Awesome!! I am happy you solved the problem!!

**Reply by siddikulakbartp** [reply]
I have the same issue. Mindray bc 5150 connected to a pc and done same range of ip. And I get ping when ping pc to the mindray bc 5150. But when trying send data from mindray bc 5150 to pc showing a prompt which is “please check network connection”. I was tried both straight through and crossover cables. Any idea to solve this ?
